distributed/system_monitor.py

```diff
@@ -89,6 +93,24 @@ def __init__(
             for k in hostcpu_c._fields:
                 self.quantities["host_cpu." + k] = deque(maxlen=maxlen)
 
+        if monitor_gil_contention is None:
+            monitor_gil_contention = dask.config.get(
+                "distributed.admin.system-monitor.gil-contention"
+            )
+        self.monitor_gil_contention = monitor_gil_contention
+        if self.monitor_gil_contention:
+            try:
+                from gilknocker import KnockKnock
+            except ImportError:
+                warnings.warn(
+                    "Monitoring GIL requires `gilknocker` to be installed. `pip install gilknocker`"
+                )
+                self.monitor_gil_contention = False
+            else:
+                self.quantities["gil_contention"] = deque(maxlen=maxlen)
+                self._gilknocker = KnockKnock(1_000)  # TODO(miles) param interval?
```

Comment:
I guess this is the only thing I'm curious about. I wonder if our benchmark suite will notice this being on or not.

@milesgranger this might be a good opportunity to trigger your first A/B

Reply:
see https://github.com/coiled/coiled-runtime/blob/main/AB_environments/README.md